File: app/basemate/components/crazy/cell/crazy.rb
module Components
  module Crazy::Cell
    class Crazy < Component::Cell::Dynamic

      def response
        components {
          div do
            plain '{{hello}}'
            button class: "button", id: "bla", attributes: { "@click": "callApi" } do
              plain "call API"
            end
            span class: "whhaaa", attributes: { "v-for": "user in users" } do
              plain "{{user.name}}"
            end
          end
        }
      end

    end
  end
end

File: app/basemate/components/crazy/js/crazy.js
BasemateUiCore.Vue.component('crazy-cell', {
  mixins: [BasemateUiCore.componentMixin],
  data: function data() {
    return {
      hello: "world",
      users: []
    };
  },
  mounted: function(){
    var self = this;
    setTimeout(function () {
      self.rerender();
    }, 2000);
  },
  methods: {
    callApi: function() {
      var self = this;
      BasemateUiCore.axios.get("/api/users.json").then(function(response){
        self.users = response.data;
      });
    }
  }
});
